form/data_mappers.rst

```diff
@@ -0,0 +1,208 @@
+
+Data mappers are responsible for reading and writing data from and into parent forms.
+
+The main built-in data mapper uses the :doc:`PropertyAccess component </components/property_access>`
+and will fit most cases. However, you can create your implementation that
+could, for example, pass submitted data to immutable objects via their constructor.
+
+The Difference between Data Transformers and Mappers
+----------------------------------------------------
+
+It is important to know the difference between
+:doc:`data transformers </form/data_transformers>` and mappers.
+
+* **Data transformers** change the representation of a value. E.g. from
+  ``"2016-08-12"`` to a ``DateTime`` instance;
+* **Data mappers** map data (e.g. an object or array) to form fields, and vice versa.
+
+Changing a ``YYYY-mm-dd`` string value to a ``DateTime`` instance is done by a
+data transformer. Populating inner fields (e.g year, hour, etc) of a compound date type using a ``DateTime`` instance is done by the data mapper.
+
+Creating a Data Mapper
+----------------------
+
+Suppose that you want to save a set of colors to the database. For this, you're
+using an immutable color object::
+
+    // src/App/Painting/Color.php
+    namespace App\Painting;
+
+    final class Color
+    {
+        private $red;
+        private $green;
+        private $blue;
+
+        public function __construct(int $red, int $green, int $blue)
+        {
+            $this->red = $red;
+            $this->green = $green;
+            $this->blue = $blue;
+        }
+
+        public function getRed(): int
+        {
+            return $this->red;
+        }
+
+        public function getGreen(): int
+        {
+            return $this->green;
+        }
+
+        public function getBlue(): int
+        {
+            return $this->blue;
+        }
+    }
+
+The form type should be allowed to edit a color. But because you've decided to
+make the ``Color`` object immutable, a new color object has to be created each time
+one of the values is changed.
+
+.. tip::
+
+    If you're using a mutable object with constructor arguments, instead of
+    using a data mapper, you should configure the ``empty_data`` with a closure
+    as described in
+    :ref:`How to Configure empty Data for a Form Class <forms-empty-data-closure>`.
+
+The red, green and blue form fields have to be mapped to the constructor
+arguments and the ``Color`` instance has to be mapped to red, green and blue
+form fields. Recognize a familiar pattern? It's time for a data mapper!
+
+.. code-block:: php
+
+    // src/App/Form/DataMapper/ColorMapper.php
+    namespace App\Form\DataMapper;
+
+    use App\Painting\Color;
+    use Symfony\Component\Form\DataMapperInterface;
+    use Symfony\Component\Form\Exception\UnexpectedTypeException;
+    use Symfony\Component\Form\FormInterface;
+
+    final class ColorMapper implements DataMapperInterface
+    {
+        /**
+         * {@inheritdoc}
```

Comment:
I think we should drop this.

Reply:
Agree. @HeahDude , I searched  throughout the docs and found only 2 `{@inheritdoc}` tags. Although I would definitely put them in my project, here in docs they make examples unnecessarily verbose.